### PR TITLE
Purge source files after 30 days by default.

### PR DESCRIPTION
**Pull request by petterreinholdtsen**
Directories are not purged, not sure why.  And the purge message do not show up in the log for some reason.

### CODE REVIEW

**Comment by petterreinholdtsen** [comment]
Forgot to mention that this is a fix fur issue #21.

**Comment by petterreinholdtsen** [comment]
I figured out the problems with directories and log messages, and the updated patch fixes all known issues related to removal of old files.

File: bin/tsdfx/copy.c [hunk 1]
```diff
@@ -273,6 +292,26 @@ tsdfx_copy_delete(struct tsd_task *t)
...
+tsdfx_copy_purgesource_child(void *ud)
+{
+	struct tsdfx_copy_task_data *ctd = ud;
+
+	/*
+	 * At this point, running as the owner of the file.
+	 */
+	if (-1 == remove(ctd->src)) {
+		/*
+		 * FIXME: this message do not make it into the log.
```
Comment (dag-erling) [inline bin/tsdfx/copy.c:307]: The task doesn't have `TASK_STDERR_PIPE` set, so the child's `stderr` goes to wherever the parent's goes (`/dev/null` when daemonized). You need to set `TASK_STDERR_PIPE` and have the parent read the output like we do for scan tasks, or call either `tsdfx_log_init()` or `tsdfx_log_userlog()` and use the appropriate macro.

File: bin/tsdfx/copy.c [hunk 2]
```diff
@@ -273,6 +292,26 @@ tsdfx_copy_delete(struct tsd_task *t)
...
+		 */
+		fprintf(stderr, "failed to purge old source file %s: %s",
```
Comment (dag-erling) [inline bin/tsdfx/copy.c]: Missing newline.

**Pull request merged**